SDWebImage/SDWebImageDownloader.m

```diff
@@ -96,7 +99,7 @@ - (nonnull instancetype)initWithSessionConfiguration:(nullable NSURLSessionConfi
 #else
         _HTTPHeaders = [@{@"Accept": @"image/*;q=0.8"} mutableCopy];
 #endif
-        _barrierQueue = dispatch_queue_create("com.hackemist.SDWebImageDownloaderBarrierQueue", DISPATCH_QUEUE_CONCURRENT);
+        _lock = dispatch_semaphore_create(1);
```

Comment:
Maybe make this "operationsLock" and `LOCK(operationsLock)`, in case there would be any other locks needed

Reply:
:) OK...